Sources/Sentry/SentryClient.m

```diff
@@ -80,11 +82,22 @@ - (_Nullable instancetype)initWithOptions:(SentryOptions *)options {
 
 - (SentryTransport *)transport {
     if (_transport == nil) {
-        _transport = [[SentryTransport alloc] initWithOptions:self.options];
+        NSError* error = nil;
+        SentryFileManager *fileManager = [[SentryFileManager alloc] initWithDsn:_options didFailWithError:&error];
+        if (nil != error) {
+            [SentryLog logWithMessage:(error).localizedDescription andLevel:kSentryLogLevelError];
+            return nil;
+        }
+        self.fileManager = fileManager;
+        _transport = [[SentryTransport alloc] initWithOptions:self.options sentryFileManager:fileManager];
```

Comment:
This will for sure conflict with @philipphofmann 😅

Reply:
dang, can we merge @philipphofmann after my deadline? 👯‍♂